channels_redis/core.py

```diff
@@ -487,18 +497,34 @@ def connection(self, index):
         if not 0 <= index < self.ring_size:
             raise ValueError("There are only %s hosts - you asked for %s!" % (self.ring_size, index))
         # Make a context manager
-        return self.ConnectionContextManager(self.hosts[index])
+        return self.ConnectionContextManager(self.hosts[index], index)
 
     class ConnectionContextManager:
         """
         Async context manager for connections
         """
-        def __init__(self, kwargs):
+        # a static member dictionary object (single instance for all the instances of the context)
+        # maps loops and indexes to redis connection pools
+        connection_dict = {}
+
+        def __init__(self, kwargs, index):
             self.kwargs = kwargs
+            self.index = index
 
         async def __aenter__(self):
-            self.conn = await aioredis.create_redis(**self.kwargs)
-            return self.conn
+            # get connection key for the current loop and the given index
+            connection_key = asyncio.get_event_loop().__hash__() + self.index
```

Comment:
Why hash rather than ID? I would say ID makes more sense since it's an object instance, unless event loops implemented the hash function when I wasn't looking!

Reply:
Yup.agreed. will change